### PR TITLE
Use netcat to detect when Kafka is up

### LINKED ISSUE #1004: Fix flaky Kafka integration test
**TestKafkaStorage/GetTrace** fails and causes the build to fail. Update the test to be reliable. 

```
--- FAIL: TestKafkaStorage (6.18s)
    --- FAIL: TestKafkaStorage/GetTrace (3.02s)
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 1 out of 30
    	integration_test.go:180: Trace was not found
    	. 
        .
        .
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 30 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:180: Trace was not found
    	assertions.go:256: 
    			Error Trace:	integration_test.go:184
    			            				integration_test.go:377
    			Error:      	Should be true
    			Test:       	TestKafkaStorage/GetTrace
    	assertions.go:256: 
    			Error Trace:	domain_trace_compare_test.go:50
    			            				integration_test.go:185
    			            				integration_test.go:377
    			Error:      	Expected value not to be nil.
    			Test:       	TestKafkaStorage/GetTrace
```

### PR DESCRIPTION
**Pull request by isaachier**
## Which problem is this PR solving?
- Fixes #1004 (hopefully). 

## Short description of the changes
- Use netcat to wait until Kafka is up before beginning integration test.

### CODE REVIEW

File: scripts/travis/kafka-integration-test.sh [hunk 1]
```diff
@@ -4,6 +4,14 @@ set -e
 
 docker pull spotify/kafka
 CID=$(docker run -d -p 2181:2181 -p 9092:9092 --env ADVERTISED_HOST=localhost --env ADVERTISED_PORT=9092 --rm spotify/kafka)
+
+# Guarantees no matter what happens, docker will remove the instance at the end.
+trap 'docker rm -f $CID 2>/dev/null' EXIT INT TERM
+
 export STORAGE=kafka
+while true; do
+    if nc -z localhost 9092; then
```
Comment (jpkrohling) [inline scripts/travis/kafka-integration-test.sh]: Does it fail immediately if the port is not up, or does it block for a few ms/seconds?

Comment (yurishkuro) [reply scripts/travis/kafka-integration-test.sh]: Looks like it blocks forever until port is available. The build still failed though.

Comment (isaachier) [reply scripts/travis/kafka-integration-test.sh]: It claims to "Specifies that nc should just scan for listening daemons, without sending any data to them." (https://linux.die.net/man/1/nc). I was checking source to see details, but will get back to you on that. Meanwhile, will investigate build issue.

Comment (isaachier) [reply scripts/travis/kafka-integration-test.sh]: Build didn't fail. @black-adder canceled every test but the kafka test and it has succeeded at least once. He said he would retry it a few more times to guarantee it works.

Comment (isaachier) [reply scripts/travis/kafka-integration-test.sh]: From a practical viewpoint, when I run `nc -z localhost 9092` on my MacBook, and Kafka is not running, the command returns within 10 ms (using `time` on 100 calls in a row gives about 500 ms).

**Comment by black-adder** [comment]
I ran the kafka integration test 7 times in a row and all passed. I think this is solid.

**Pull request merged**